R/dash.R

```diff
@@ -214,19 +217,14 @@ Dash <- R6::R6Class(
           return(FALSE)
         }
 
-        # client wants the mapping formatted this way -- https://github.com/plotly/dash/blob/d2ebc837/dash/dash.py#L367-L378
-        outputs <- strsplit(names(private$callback_map), "\\.")
-        payload <- Map(function(x, y) {
-          # IMPORTANT: if state/events don't exist, dash-renderer wants them
-          # to be an empty array (i.e., null/missing won't work)
+        payload <- Map(function(callback_signature) {
           list(
-            output = list(id = y[[1]], property = y[[2]]),
-            inputs = setNames(callback_inputs(x), NULL),
-            state = setNames(callback_states(x), NULL),
-            # Chris mentioned that events might/should be deprecated
-            events = setNames(callback_events(x), NULL)
+            output=callback_signature$output,
+            inputs=callback_signature$inputs,
+            state=callback_signature$state,
+            key=paste(callback_signature$output$id, callback_signature$output$property, sep='.')
```

Comment:
Is `key` a special keyword in R's lists? I'm assuming so, since I don't see this referenced anywhere else in the code.

Reply:
This was present in the original code, but I can remove it. The key is also referenced here:

https://github.com/plotly/dashR/blob/b05eaed4ed77ce625ce7e794e2bc80aa622cd640/R/dependencies.R#L55

Reply:
These `key=` lines are still present 🔪

Reply:
fixed in 7e51f91b2dc11155043c5957cf527d568c8f66b0

Reply:
fixed in 313dff1a9c99826fa05bea285fec73ac49e59169